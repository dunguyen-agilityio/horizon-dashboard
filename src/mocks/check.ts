import { TCheck } from '@/types/check';

export const MOCK_CHECKS: TCheck[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    quantity: 28,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    quantity: 30,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
