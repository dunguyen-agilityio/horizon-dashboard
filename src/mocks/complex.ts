import { Complex, ComplexStatus } from '@/types/complex';

export const MOCK_COMPLEXES: Complex[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    status: ComplexStatus.DISABLE,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    status: ComplexStatus.ERROR,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    status: ComplexStatus.APPROVED,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
