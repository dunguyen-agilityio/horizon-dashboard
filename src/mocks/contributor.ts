import { Contributor } from '@/models/Contributor';
import { MOCK_USERS } from './user';

export const MOCK_CONTRIBUTORS = [
  {
    ...MOCK_USERS[0],
    rating: 23,
    template: 'Abstract Colors',
    createdAt: new Date('2024-01-13T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[1],
    rating: 80,
    template: 'Swipe Circles',
    createdAt: new Date('2024-08-13T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[2],
    template: 'Colorful Heaven',
    createdAt: new Date('2024-08-11T07:25:04.188Z'),
  },
  {
    ...MOCK_USERS[3],
    template: '3D Cubes Art',
    avatar: undefined,
  },
] as Contributor[];
