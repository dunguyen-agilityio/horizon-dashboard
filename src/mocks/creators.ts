export const MOCK_CREATORS = [
  {
    id: '1',
    avatar: '/default-avatar.webp',
    userName: 'maddison_c21',
    artwork: 9821,
    rating: 90,
  },
  {
    id: '2',
    avatar: '/default-avatar.webp',
    userName: 'karl.willO2',
    artwork: 7032,
    rating: 80,
  },
  {
    id: '3',
    avatar: '/default-avatar.webp',
    userName: 'andreea.1z',
    artwork: 5204,
    rating: 60,
  },
];
