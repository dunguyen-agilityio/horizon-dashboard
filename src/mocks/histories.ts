import { NFT_IMAGES } from '@/constants/images';

export const histories = [
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'Colorful Heaven',
    author: 'Mark Benjamin',
    price: '1.33',
    timeAgo: new Date(Date.now() - 2 * 60 * 60 * 1000),
  },
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'Abstract Colors',
    author: 'Esthera Jackson',
    price: '0.91',
    timeAgo: new Date(Date.now() - 6 * 60 * 60 * 1000),
  },
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'ETH AI Brain',
    author: 'Nick Wilson',
    price: '2.82',
    timeAgo: new Date(Date.now() - 5 * 60 * 60 * 1000),
  },
  {
    image: NFT_IMAGES.DEFAULT,
    title: 'Swipe Circles',
    author: 'Peter Will',
    price: '2.32',
    timeAgo: new Date(Date.now() - 1.5 * 24 * 60 * 60 * 1000),
  },
];
