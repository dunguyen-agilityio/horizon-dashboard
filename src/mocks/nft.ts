// Types
import { NFTResponse, type NFTData } from '@/models/NFT';

// Constants
import { NFT_IMAGES } from '@/constants/images';

// Mocks
import { MOCK_IMAGE_RESPONSE, MOCK_USERS, MOCK_USERS_RESPONSE } from './user';
import { StrapiResponse } from '@/types/strapi';

const [MOCK_USER] = MOCK_USERS;

export const MOCK_NFTS: NFTData[] = [
  {
    id: '1',
    name: 'Mesh Gradients',
    author: MOCK_USER,
    price: 1.23,
    image: { url: NFT_IMAGES.DEFAULT, hash: 'mock hash' },
    members: MOCK_USERS,
    createdAt: new Date('2024-08-17T07:25:04.188Z'),
  },
];

export const MOCK_NFTS_RESPONSE: StrapiResponse<NFTResponse>[] = [
  {
    id: '1',
    attributes: {
      name: 'Mesh Gradients',
      author: { data: MOCK_USERS_RESPONSE[0] },
      price: 1.23,
      members: { data: MOCK_USERS_RESPONSE },
      image: { data: MOCK_IMAGE_RESPONSE },
      createdAt: new Date('2024-08-17T07:25:04.188Z'),
    },
  },
];
