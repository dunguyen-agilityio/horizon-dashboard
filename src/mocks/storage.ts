export const MOCK_STORAGE = {
  usedSize: 25.6,
  totalSize: 50,
  valueProgress: 60,
};
