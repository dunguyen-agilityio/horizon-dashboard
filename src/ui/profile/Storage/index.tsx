'use client';

import { Progress } from '@nextui-org/react';

// icons
import { More, Cloud } from '@/icons';

// components
import { BoxIcon, Text } from '@/components';

interface StorageProps {
  usedSize: number;
  totalSize: number;
  valueProgress: number;
}

const Storage = ({ usedSize, totalSize, valueProgress }: StorageProps) => (
  <div className="flex flex-col h-[365px] bg-white dark:bg-indigo p-5 rounded-md">
    <div className="flex justify-end">
      <BoxIcon
        icon={<More />}
        customClass="rotate-90 w-9 h-9 bg-gray dark:bg-indigo-light rounded-lg fill-blue-450 dark:fill-white"
      />
    </div>
    <div className="flex flex-col items-center pb-12">
      <div className="w-[100px] h-[100px] rounded-full bg-gray dark:bg-indigo-light flex items-center justify-center mb-3">
        <BoxIcon
          icon={<Cloud />}
          customClass="fill-indigo-light w-11 h-8 rounded-lg fill-blue-450 dark:fill-white"
        />
      </div>
      <Text className="font-bold text-[24px]">Your Storage</Text>
      <Text className="font-normal text-secondary text-center w-2/3 leading-relaxed">
        Supervise your drive space in the easiest way
      </Text>
    </div>

    <div>
      <div className="flex justify-between gap-1">
        <Text className="font-normal text-secondary">{usedSize}Gb</Text>
        <Text className="font-normal text-secondary">{totalSize}Gb</Text>
      </div>
      <Progress aria-label="Storage" value={valueProgress} />
    </div>
  </div>
);

export default Storage;
