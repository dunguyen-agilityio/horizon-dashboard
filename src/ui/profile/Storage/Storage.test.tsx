import { render, screen } from '@testing-library/react';

// mocks
import { MOCK_STORAGE } from '@/mocks/storage';

import Storage from '.';

describe('Storage component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Storage {...MOCK_STORAGE} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the component correctly', () => {
    render(<Storage {...MOCK_STORAGE} />);

    expect(screen.getByText('Your Storage')).toBeInTheDocument();
    expect(
      screen.getByText('Supervise your drive space in the easiest way'),
    ).toBeInTheDocument();
    expect(screen.getByText('25.6Gb')).toBeInTheDocument();
    expect(screen.getByText('50Gb')).toBeInTheDocument();
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });
});
