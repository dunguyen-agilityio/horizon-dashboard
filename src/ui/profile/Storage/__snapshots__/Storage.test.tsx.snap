// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storage component should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col w-[381px] h-[365px] bg-white dark:bg-indigo p-5 rounded-md"
  >
    <div
      class="flex justify-end"
    >
      <i
        class="flex justify-center items-center hover:cursor-pointer rotate-90 w-9 h-9 bg-gray dark:bg-indigo-light rounded-lg fill-blue-450 dark:fill-white"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_201_3969)"
          >
            <path
              d="M12 8C13.1 8 14 7.1 14 6C14 4.9 13.1 4 12 4C10.9 4 10 4.9 10 6C10 7.1 10.9 8 12 8ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10ZM12 16C10.9 16 10 16.9 10 18C10 19.1 10.9 20 12 20C13.1 20 14 19.1 14 18C14 16.9 13.1 16 12 16Z"
            />
          </g>
          <defs>
            <clippath
              id="clip0_201_3969"
            >
              <rect
                fill="white"
                height="24"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
      </i>
    </div>
    <div
      class="flex flex-col items-center pb-12"
    >
      <div
        class="w-[100px] h-[100px] rounded-full bg-gray dark:bg-indigo-light flex items-center justify-center mb-3"
      >
        <i
          class="flex justify-center items-center hover:cursor-pointer w-11 h-8 rounded-lg fill-blue-450 dark:fill-white"
        >
          <svg
            height="32"
            viewBox="0 0 46 32"
            width="46"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M37.0875 12.2434C35.7842 5.63085 29.9767 0.666687 23 0.666687C17.4608 0.666687 12.65 3.81002 10.2542 8.41002C4.485 9.02335 0 13.9109 0 19.8334C0 26.1775 5.15583 31.3334 11.5 31.3334H36.4167C41.7067 31.3334 46 27.04 46 21.75C46 16.69 42.0708 12.5884 37.0875 12.2434ZM36.4167 27.5H11.5C7.26417 27.5 3.83333 24.0692 3.83333 19.8334C3.83333 15.9042 6.76583 12.6267 10.6567 12.2242L12.7075 12.0134L13.6658 10.1925C15.4867 6.68502 19.0517 4.50002 23 4.50002C28.0217 4.50002 32.3533 8.06502 33.3308 12.9909L33.9058 15.8659L36.8383 16.0767C39.8283 16.2684 42.1667 18.7792 42.1667 21.75C42.1667 24.9125 39.5792 27.5 36.4167 27.5ZM19.1667 20.1784L15.1608 16.1725L12.4583 18.875L19.1667 25.5834L30.6858 14.0642L27.9833 11.3617L19.1667 20.1784Z"
              fill="#4318ff"
            />
          </svg>
        </i>
      </div>
      <p
        class="text-primary dark:text-white leading-xl font-bold text-[24px]"
      >
        Your Storage
      </p>
      <p
        class="text-primary dark:text-white text-medium leading-xl font-normal text-secondary text-center w-2/3 leading-relaxed"
      >
        Supervise your drive space in the easiest way
      </p>
    </div>
    <div>
      <div
        class="flex justify-between gap-1"
      >
        <p
          class="text-primary dark:text-white text-medium leading-xl font-normal text-secondary"
        >
          25.6
          Gb
        </p>
        <p
          class="text-primary dark:text-white text-medium leading-xl font-normal text-secondary"
        >
          50
          Gb
        </p>
      </div>
      <div
        aria-label="Storage"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="60"
        aria-valuetext="60%"
        class="flex flex-col gap-2 w-full"
        id="react-aria-:r0:"
        role="progressbar"
      >
        <div
          class="z-0 relative bg-default-300/50 overflow-hidden h-3 rounded-full"
        >
          <div
            class="h-full bg-primary rounded-full transition-transform !duration-500"
            style="transform: translateX(-100%);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
